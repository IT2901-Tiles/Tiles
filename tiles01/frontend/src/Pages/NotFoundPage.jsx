import React from 'react';
import LinkButton from '../Components/LinkButton';
import '../CSS/PageNotFound.css';
import bikeCard from "../images/things/bike.png";
import colorChangeCard from "../images/feedbacks/color_change.png";
import motionCard from "../images/triggers/motion.png"
import dropCard from "../images/triggers/drop.png"
import penCard from "../images/things/pen.png";
import temperatureCard from "../images/triggers/temperature.png"
import umbrellaCard from "../images/things/umbrella.png"
import vibrateCard from "../images/feedbacks/vibrate.png"; 

function NotFoundPage() {

    return (
        <div className="PageNotFound">
            <div className="pageNotFoundTitle">
                <h1>Page not found</h1>
            </div>

            <div className="ImagesPageNotFound">
                {/* Left side of the 404-page */}
            <img src={bikeCard} alt="TheBikeCard" className="image bikeCard"/>
            <img src={colorChangeCard} alt="Card" className="image colorChangeCard"/>
            <img src={motionCard} alt="A card" className="image motionCard"/>
            <img src={dropCard} alt="a card" className="image dropCard"/>
            <img src={penCard} alt="a card" className="image penCard"/>

                {/* Right side of the 404-page */}
            {/* <img src={colorChangeCard} alt="a card" className="image colorChangeCard2"/> */}
            <img src={temperatureCard} alt="a card" className="image temperatureCard"/>
            <img src={umbrellaCard} alt="a card" className="image umbrellaCard"/>
            <img src={vibrateCard} alt="a card" className="image vibrateCard"/>
            <img src={motionCard} alt="a card" className="image motionCard2"/>
            <img src={dropCard} alt="a card" className="image dropCard2"/>
            </div>

            <div className="middleContainer">
                <p>It seems like you are a bit lost. 
                    Let me help you find your way back! </p>
            </div>

            <div className="bottomContainer">
            <div className="cardsButton"> 
             <LinkButton title="Back to the cards" target="/cards" size="wider Large"></LinkButton>
            </div>
            <div className="mainPageButton"> 
              <LinkButton title="Main page" target="/" size="wider Large"></LinkButton>
            </div>
            </div>
        </div>

    );
}

export default NotFoundPage;