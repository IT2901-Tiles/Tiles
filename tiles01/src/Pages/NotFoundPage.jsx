import React from 'react';
import LinkButton from '../Components/LinkButton';
import '../CSS/PageNotFound.css';
//IMG imports
import bikeCard from "../images/things/bike.png";
import colorChangeCard from "../images/feedbacks/color_change.png";
import motionCard from "../images/triggers/motion.png"
import dropCard from "../images/triggers/drop.png"
import penCard from "../images/things/pen.png";
import shakeCard from "../images/triggers/temperature.png"
import umbrellaCard from "../images/things/umbrella.png"
import vibrateCard from "../images/feedbacks/vibrate.png"; 
//This page does not exist
function NotFoundPage() {
    return (
        <div className="PageNotFound">
            <div className="pageNotFoundTitle">
                <h1>Page not found</h1>
            </div>

            <img src={bikeCard} alt="TheBikeCard" className="image bikeCard"/>
            <img src={colorChangeCard} alt="Card" className="image colorChangeCard"/>
            <img src={motionCard} alt="A card" className="image motionCard"/>
            <img src={dropCard} alt="a card" className="image dropCard"/>
            <img src={penCard} alt="a card" className="image penCard"/>

            <img src={colorChangeCard} alt="a card" className="image colorChangeCard2"/>
            <img src={shakeCard} alt="a card" className="image shakeCard"/>
            <img src={umbrellaCard} alt="a card" className="image umbrellaCard"/>
            <img src={vibrateCard} alt="a card" className="image vibrateCard"/>
            <img src={motionCard} alt="a card" className="image motionCard2"/>
            <img src={dropCard} alt="a card" className="image dropCard2"/>

            <div className="middleContainer">
                <p>It seems like you are a bit lost. 
                    Let me help you find your way back! </p>
            </div>

            <div className="bottomContainer">
            <div className="cardsButton"> 
             <LinkButton title="Back to the cards" target="/cards" size="wider Large"></LinkButton>
            </div>
            <div className="mainPageButton"> 
              <LinkButton title="Main page" target="/" size="wider Large"></LinkButton>
            </div>
            </div>
        </div>
    );
}

export default NotFoundPage;