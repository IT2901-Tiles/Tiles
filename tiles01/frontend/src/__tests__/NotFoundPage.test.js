import {configure, shallow} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import NotFoundPage from '../Pages/NotFoundPage';
import {cleanup} from '@testing-library/react';

afterEach(() => {
    cleanup();
});

configure({
    adapter: new Adapter()
});

describe('Not Found Page', () => {
    const wrapper = shallow(<NotFoundPage/>)
    it('renders correctly', () => {
        shallow(<NotFoundPage/>);
    });
    it('renders two LinkButton elements', () => {
        expect(wrapper.find('LinkButton')).toHaveLength(2);
    });
    it('renders the p element correct', () => {
        const p = wrapper.find('p');
        expect(p).toHaveLength(1);
        expect(p.text()).toEqual('It seems like you are a bit lost. Let me help you find your way back! ');
    });
    it('renders the h1 element correct', () => {
        const heading = wrapper.find('h1');
        expect(heading).toHaveLength(1);
        expect(heading.text()).toEqual('Page not found');
    });
    it('renders eleven images', () => {
        expect(wrapper.find('img')).toHaveLength(11);
    });
});