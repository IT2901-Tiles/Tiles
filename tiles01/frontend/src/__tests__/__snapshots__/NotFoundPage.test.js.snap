// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page renders correctly 1`] = `
<div
  className="PageNotFound"
>
  <div
    className="pageNotFoundTitle"
  >
    <h1>
      Page not found
    </h1>
  </div>
  <img
    alt="TheBikeCard"
    className="image bikeCard"
    src="bike.png"
  />
  <img
    alt="Card"
    className="image colorChangeCard"
    src="color_change.png"
  />
  <img
    alt="A card"
    className="image motionCard"
    src="motion.png"
  />
  <img
    alt="a card"
    className="image dropCard"
    src="drop.png"
  />
  <img
    alt="a card"
    className="image penCard"
    src="pen.png"
  />
  <img
    alt="a card"
    className="image colorChangeCard2"
    src="color_change.png"
  />
  <img
    alt="a card"
    className="image temperatureCard"
    src="temperature.png"
  />
  <img
    alt="a card"
    className="image umbrellaCard"
    src="umbrella.png"
  />
  <img
    alt="a card"
    className="image vibrateCard"
    src="vibrate.png"
  />
  <img
    alt="a card"
    className="image motionCard2"
    src="motion.png"
  />
  <img
    alt="a card"
    className="image dropCard2"
    src="drop.png"
  />
  <div
    className="middleContainer"
  >
    <p>
      It seems like you are a bit lost. Let me help you find your way back! 
    </p>
  </div>
  <div
    className="bottomContainer"
  >
    <div
      className="cardsButton"
    >
      <a
        href="/cards"
        onClick={[Function]}
      >
        <button
          className="wider LargeButton btn btn-primary"
          disabled={false}
          style={
            Object {
              "backgroundColor": "#F08A00",
              "color": "#F3EAC2",
            }
          }
          type="button"
        >
          Back to the cards
        </button>
      </a>
    </div>
    <div
      className="mainPageButton"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="wider LargeButton btn btn-primary"
          disabled={false}
          style={
            Object {
              "backgroundColor": "#F08A00",
              "color": "#F3EAC2",
            }
          }
          type="button"
        >
          Main page
        </button>
      </a>
    </div>
  </div>
</div>
`;
